src/main/java/com/imsweb/datagenerator/naaccr/rule/patient/BirthRule.java

```diff
@@ -22,13 +26,30 @@ public BirthRule() {
     }
 
     @Override
-    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options) {
+    public void execute(Map<String, String> record, List<Map<String, String>> otherRecords, NaaccrDataGeneratorOptions options, Map<String, Object> context) {
 
         // birth date should be no later than five years prior to min dx date (or current date if min dx date not defined)
-        LocalDate maxBirthDate = options == null ? LocalDate.now().minusYears(15) : options.getMinDxDate().minusYears(5);
+        //LocalDate maxBirthDate = options == null ? LocalDate.now().minusYears(15) : options.getMinDxDate().minusYears(5);
+        LocalDate maxBirthDate = options == null ? LocalDate.now() : options.getMinDxDate().minusYears(5);
         // limit age to max 100 years
         LocalDate minBirthDate = maxBirthDate.minusYears(100);
 
+        // Based on our tumors, pick an age that's possible for these sites.
+        Integer maxAgeGroup = (Integer)context.get(CONTEXT_FLAG_MAX_AGE_GROUP);
+        if (maxAgeGroup != null)
+            if (maxAgeGroup >= 0) {
+                maxBirthDate = minBirthDate.plusYears(100 - (maxAgeGroup * 10));
+                if (options != null) {
+                    LocalDate minDxDate = options.getMinDxDate();
+                    LocalDate maxDxDate = options.getMaxDxDate();
+                    long daysBetween = minDxDate.until(maxDxDate, DAYS) - 1;
+                    if (daysBetween > 3650) daysBetween = 3650;
```

Comment:
Now I am sure you are not formatting your files; the assignment shouldn't be on the same line as the if statement :-)  I am going to stop reporting these, please make sure review all your changes, and maybe to turn the auto-formatting on.

Reply:
Also, that 3650 hard-coded value is difficult to understand. Can you please add a one sentence comment on the top of the if statement describing what you are doing and what that value represents? That will help the readability.